perfkitbenchmarker/linux_benchmarks/cloudsuite_graph_analytics_benchmark.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs the graph analytics benchmark of Cloudsuite.
+
+More info: http://cloudsuite.ch/graphanalytics/
+"""
+
+import re
+
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import sample
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.linux_packages import docker
+
+flags.DEFINE_integer('cloudsuite_graph_analytics_worker_mem',
+                     2,
+                     'Amount of memory for the worker, in gigabytes,'
+                     'use 0 to ignore this option')
+
+FLAGS = flags.FLAGS
+
+BENCHMARK_NAME = 'cloudsuite_graph_analytics'
+BENCHMARK_CONFIG = """
+cloudsuite_graph_analytics:
+  description: >
+    Run Cloudsuite graph analytics benchmark. Specify the number of worker
+    VMs with --num_vms.
+  vm_groups:
+    master:
+      vm_spec: *default_single_core
+      vm_count: 1
+    workers:
+      vm_spec: *default_single_core
+"""
+
+
+def GetConfig(user_config):
+  """ Reads the config file and spawns overwrites the  number of worker vms
+  config file option with num_vms
+  """
+  config = configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+  if FLAGS['num_vms'].present:
+    config['vm_groups']['workers']['vm_count'] = FLAGS.num_vms
+  return config
+
+
+def Prepare(benchmark_spec):
+  """Install docker. Pull the required images from DockerHub. Create datasets.
+  Start Spark master and workers.
```

Comment:
Similar style comment here. One-line summary first. If necessary, follow with a blank line and then more details.